diff -Naur a/hardware/broadcom/libbt/Android.mk b/hardware/broadcom/libbt/Android.mk
--- a/hardware/broadcom/libbt/Android.mk	2014-03-09 19:29:36.499262424 -0400
+++ b/hardware/broadcom/libbt/Android.mk	2014-02-23 00:51:20.310642000 -0500
@@ -1,6 +1,7 @@
 LOCAL_PATH := $(call my-dir)
 
 ifneq ($(BOARD_HAVE_BLUETOOTH_BCM),)
+ifneq ($(TARGET_DEVICE),m470)
 
 include $(CLEAR_VARS)
 
@@ -45,3 +46,4 @@
 endif
 
 endif # BOARD_HAVE_BLUETOOTH_BCM
+endif
diff -Naur a/hardware/broadcom/wlan/bcmdhd/wpa_supplicant_8_lib/Android.mk b/hardware/broadcom/wlan/bcmdhd/wpa_supplicant_8_lib/Android.mk
--- a/hardware/broadcom/wlan/bcmdhd/wpa_supplicant_8_lib/Android.mk	2014-03-17 13:24:49.298475973 -0400
+++ b/hardware/broadcom/wlan/bcmdhd/wpa_supplicant_8_lib/Android.mk	2014-03-15 14:45:07.002604000 -0400
@@ -14,6 +14,7 @@
 # limitations under the License.
 #
 LOCAL_PATH := $(call my-dir)
+ifneq ($(TARGET_DEVICE),m470)
 
 ifeq ($(WPA_SUPPLICANT_VERSION),VER_0_8_X)
 
@@ -65,3 +66,4 @@
 ########################
 
 endif
+endif
