diff -Naur a/hardware/broadcom/libbt/Android.mk b/hardware/broadcom/libbt/Android.mk
--- a/hardware/broadcom/libbt/Android.mk	2014-03-09 19:29:36.499262424 -0400
+++ b/hardware/broadcom/libbt/Android.mk	2014-02-23 00:51:20.310642000 -0500
@@ -1,6 +1,7 @@
 LOCAL_PATH := $(call my-dir)
 
 ifneq ($(BOARD_HAVE_BLUETOOTH_BCM),)
+ifneq ($(TARGET_DEVICE),m470)
 
 include $(CLEAR_VARS)
 
@@ -45,3 +46,4 @@
 endif
 
 endif # BOARD_HAVE_BLUETOOTH_BCM
+endif
diff -Naur a/hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/device-bcm.mk b/hardware/broadcom/wlan/bcmdhd/firmware/device-bcm.mk
--- a/hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/device-bcm.mk	2014-03-14 09:40:57.003218000 -0400
+++ b/hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/device-bcm.mk	2014-03-14 20:20:28.011053892 -0400
@@ -15,7 +15,10 @@
 #
 
 ########################
+
+ifneq ($(TARGET_DEVICE),m470)
 -include hardware/broadcom/wlan/bcmdhd/config/config-bcm.mk
+endif
 
 ifeq ($(strip $(WIFI_BAND)),802_11_BG)
 BCM_FW_SRC_FILE_STA := fw_bcm4330_bg.bin
