diff -Naur a/system/core/init/init.c b/system/core/init/init.c
--- a/system/core/init/init.c	2014-06-13 09:54:34.309405000 -0400
+++ b/system/core/init/init.c	2014-06-13 10:23:12.613398194 -0400
@@ -822,6 +822,14 @@
         cnt = snprintf(prop, sizeof(prop), "ro.boot.%s", boot_prop_name);
         if (cnt < PROP_NAME_MAX)
             property_set(prop, value);
+    /* DoPa (20140312) - the Hisense bootloader fails to use 'androidboot.mode=charger'
+     * to indicate that the tablet is booting solely to show the battery animations;
+     * this workaround looks for a particular value of 'power_reason' that corresponds
+     * to that state (Hisense appears to have done something similar in their version
+     * of 'init').
+     */
+    } else if (!strcmp(name, "power_reason") && !strcmp(value, "0x808800")) {
+        property_set("ro.boot.mode", "charger");
     }
 }
 
diff -Naur a/system/core/rootdir/init.rc b/system/core/rootdir/init.rc
--- a/system/core/rootdir/init.rc	2014-06-13 09:54:34.373405000 -0400
+++ b/system/core/rootdir/init.rc	2014-06-13 10:23:35.873398089 -0400
@@ -34,7 +34,7 @@
 
 sysclktz 0
 
-loglevel 3
+loglevel 6
 
 # Backward compatibility
     symlink /system/etc /etc
