diff -Naur a/external/wpa_supplicant_8/Android.mk b/external/wpa_supplicant_8/Android.mk
--- a/external/wpa_supplicant_8/Android.mk	2014-06-13 09:41:09.037409000 -0400
+++ b/external/wpa_supplicant_8/Android.mk	2014-06-13 10:07:31.521402433 -0400
@@ -1,5 +1,7 @@
 LOCAL_PATH:= $(call my-dir)
 
+ifneq ($(TARGET_DEVICE),m470)
+
 ifndef WPA_SUPPLICANT_VERSION
 WPA_SUPPLICANT_VERSION := VER_0_8_X
 endif
@@ -14,3 +16,4 @@
 ifeq ($(WPA_SUPPLICANT_VERSION),VER_2_1_DEVEL)
 include $(call all-subdir-makefiles)
 endif
+endif
